argon2/src/block.rs

```diff
@@ -101,6 +106,12 @@ impl Block {
         q ^= &r;
         q
     }
+
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    #[target_feature(enable = "avx2")]
+    pub(crate) unsafe fn compress_avx2(rhs: &Self, lhs: &Self) -> Self {
+        Self::compress_soft(rhs, lhs)
+    }
```

Comment:
It feels like this is all an implementation detail that could be factored somewhere else, e.g. nested inner functions

Reply:
Did you have something specific in mind? I'm not quite seeing what you're saying.

Reply:
I can take care of it in a followup